### ISSUE TITLE
Request: ability to set all IDs to UID

### ISSUE THREAD

**Issue by JerBlo** [SSilence/php-imap-client#148]
I would like to have the ability to set a configuration parameter to ImapClient, so all IDs used in public and private functions are in UID.

We need this for a tool where multiple clients read from the same IMAP client at the same time. Deleting a message would change the normal message number, which can cause mixing emails. Also, when processing e-mails in a non DESC order, you need to request the ID each time based on the UID to be sure the right ID is used.

If you like the idea and I can help implementing, let me know

**Comment by sergey144010** [comment]
Good idea, I'm not against help, but you need to wait until they accept the last pool request, as there are some changes.

**Comment by xXXIMMATTHEWXXx** [comment]
Sounds great! If you have any ideas or if you want to code it, feel free. :)

**Comment by xXXIMMATTHEWXXx** [comment]
I have merged the pull request.

**Comment by JerBlo** [comment]
Since I'm not an experienced GIT-user yet, it will take some time before able to contribute, but I'll keep this item in mind. I like the project, so love to contribute one day.

**Comment by xXXIMMATTHEWXXx** [comment]
If getting started with git is the problem all of us here can help :)

**Comment by sergey144010** [comment]
About this, need make switch in ImapClient like this
```php
$imap->useId();
$imap->useUid();
```
But need to rewrite a lot of code, and think through the logic.

**Comment by xXXIMMATTHEWXXx** [comment]
Yeah

**Comment by xXXIMMATTHEWXXx** [comment]
We need to get a move on with this @sergey144010 if you want ill do this for you
